cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/StatusToolUtils.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.replication;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hbase.util.Bytes;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Utility class for Replication Status Tool
+ */
+public final class StatusToolUtils {
+
+  private static final Log LOG = LogFactory.getLog(StatusToolUtils.class);
+  private ConcurrentHashMap<String, Long> cachedUpdates;
+
+  public StatusToolUtils() {
+    cachedUpdates = new ConcurrentHashMap<>();
+  }
+
+  /**
+   * getRowKey creates the key to be used for the replicationstate table
+   * clusterID is used to differentiate between entries from Master and Slave clusters
+   * regionName is used to identify regions
+   * rsID is used to identify regionservers because multiple regionservers can update entries for same regions
+   * @param regionName
+   * @param clusterID
+   * @param rsID
+   * @return
+   */
+  public byte[] getRowKey (String regionName, String clusterID, UUID rsID) {
+    byte[] rowKey;
+    rowKey = Bytes.add(Bytes.toBytes(Bytes.toBytes(clusterID).length), Bytes.toBytes(clusterID));
+    rowKey = Bytes.add(rowKey, Bytes.toBytes(Bytes.toBytes(regionName).length), Bytes.toBytes(regionName));
+    rowKey = Bytes.add(rowKey, Bytes.toBytes(Bytes.toBytes(rsID.toString()).length), Bytes.toBytes(rsID.toString()));
+    return (rowKey);
+  }
+
+  /**
+   * Synchronized method used by replication status tool TableUpdate class to add timestamps
+   * from coprocessors.
+   * @param regionName is the key for the Map
+   * @param time is the value for the Map
+   */
+  public synchronized void updateTime (String regionName, long time) {
+    if (!cachedUpdates.containsKey(regionName) || time > cachedUpdates.get(regionName)) {
```

Comment:
the postReplicateLogEntries() gets called in a new thread every time. I saw JVM running out of threads when postReplicateLogEntries() takes a long time to finish.